src/k2/transport/Payload.cpp

```diff
@@ -306,14 +317,18 @@ void Payload::write(const String& value) {
     write(value.data(), size);
 }
 
-void Payload::write(const std::decimal::decimal64& value) {
-    std::decimal::decimal64::__decfloat64 data = const_cast<std::decimal::decimal64&>(value).__getval();
-    write((const void*)&data, sizeof(data));
+void Payload::write(const boost::multiprecision::cpp_dec_float_50& value) {
+    PayloadWriter writer(*this);
+    skv::http::WriteStreamBuf<PayloadWriter, 80> writesb(writer);
```

Comment:
Lets add a static assert here and for dec100 below that the sizeof for them is what you have observed. That way we get a compile time error instead of a runtime error if they change the implementation to something we don't expect. Then you can keep the WriteStreamBuf implementation how you have it now. https://en.cppreference.com/w/cpp/language/static_assert